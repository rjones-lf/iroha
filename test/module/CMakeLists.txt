set(CMAKE_BUILD_TYPE Debug)

SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/test_bin)

add_subdirectory(config)
add_subdirectory(connection)
add_subdirectory(crypto)
add_subdirectory(expected)
add_subdirectory(membership_service)
#add_subdirectory(infra/repository)
#add_subdirectory(infra/service)
#add_subdirectory(infra/config)
# commented because these tests do nothing
# TODO: make them do something :)
# add_subdirectory(consensus)
#add_subdirectory(vendor)
#add_subdirectory(validation)
#add_subdirectory(transaction_builder)

add_subdirectory(service)
