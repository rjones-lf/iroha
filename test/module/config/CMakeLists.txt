# Config Utils test
add_executable(config_utils_test
  config_utils_test.cpp
)
target_link_libraries(config_utils_test
  gtest
  config_utils
  expected
  logger
)
add_test(
  NAME config_utils_test
  COMMAND $<TARGET_FILE:config_utils_test>
)
