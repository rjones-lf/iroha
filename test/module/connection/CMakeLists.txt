add_executable(connection_with_grpc_flatbuffer_test
        connection_with_grpc_flatbuffer_test.cpp
        )
target_link_libraries(connection_with_grpc_flatbuffer_test
        endpoint_fbs
        config_manager
        flatbuffer_service
        connection_with_grpc_flatbuffer
        gtest
        )
add_test(
        NAME connection_with_grpc_flatbuffer_test
        COMMAND $<TARGET_FILE:connection_with_grpc_flatbuffer_test>
)
