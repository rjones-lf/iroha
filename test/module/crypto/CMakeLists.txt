#
# Copyright Soramitsu Co., Ltd. 2016 All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

# Hash Test
add_executable(hash_test
        hash_test.cpp
)
target_link_libraries(hash_test
  hash
  crypto
  hash
  ssl
  gtest
)
add_test(
  NAME hash_test
  COMMAND $<TARGET_FILE:hash_test>
)


# Base64 Test
add_executable(base64_test base64_test.cpp)
target_link_libraries(base64_test 
  base64
  ssl
  crypto
  gtest
)
add_test(
  NAME base64_test
  COMMAND $<TARGET_FILE:base64_test>
)


# Singature Test
add_executable(signature_test signature_test.cpp)
target_link_libraries(signature_test 
  signature
  base64
  hash
  ed25519
  gtest
)
add_test(
  NAME signature_test
  COMMAND $<TARGET_FILE:signature_test>
)
