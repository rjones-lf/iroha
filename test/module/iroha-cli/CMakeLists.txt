#
# Copyright Soramitsu Co., Ltd. All Rights Reserved.
# SPDX-License-Identifier: Apache-2.0
#

addtest(client_test client_test.cpp)
target_link_libraries(client_test
    client
    processors
    server_runner
    )
target_include_directories(client_test PUBLIC
    ${PROJECT_SOURCE_DIR}/iroha-cli
    )
