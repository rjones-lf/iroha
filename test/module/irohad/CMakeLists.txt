set(CMAKE_BUILD_TYPE Debug)

SET(EXECUTABLE_OUTPUT_PATH ${PROJECT_BINARY_DIR}/test_bin)

# Reusable tests
add_subdirectory(ametsuchi)
add_subdirectory(logger)
add_subdirectory(peer_service)
