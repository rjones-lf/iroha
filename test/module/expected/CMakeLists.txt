########################################################################################
# Expected
########################################################################################

add_executable(expected_test expected_test.cpp)
target_link_libraries(expected_test
  gtest
  expected
  flatbuffers
)

add_test(
  NAME expected_test
  COMMAND $<TARGET_FILE:expected_test>
)